error/s2n_errno.h

```diff
@@ -35,6 +35,8 @@
 #define S2N_ERR_T_INTERNAL_START (S2N_ERR_T_INTERNAL << S2N_ERR_NUM_VALUE_BITS)
 #define S2N_ERR_T_USAGE_START (S2N_ERR_T_USAGE << S2N_ERR_NUM_VALUE_BITS)
 
+#define S2N_ERR_CACHE_WILL_BLOCK -2
```

Comment:
Can we move this to s2n.h? This needs to be exposed to caller.

Reply:
I've just made an update which removes this entirely. I think it is too specific since we will have multiple asynchronous functions.

The next commit preserves the -2 return code currently in use, but recommends that customer functions return S2N_ERR_BLOCKED if they would block on IO.